wurst/objects/abilities/TrollUpgradeSpell.wurst

```diff
@@ -20,86 +19,217 @@ import TrollUnitTextConstant
     new UpgradeDefinition(UPGD_MEDALLION_COURAGE)
     ..setName(1, "Medallion Of Courage")
 
-public class TrollUpgradeSuperSubSpell extends TrollUpgradeSubSpell
-
-    construct(int id, string name, string toolTipExt, Pair<int, int> btnPos, string icon, string hotkey)
-        super(id, name, toolTipExt, btnPos, icon, hotkey)
-        this.levelReq = 9
-        this.upgradeReq = toRawCode(UPGD_MEDALLION_COURAGE)
-
-public class TrollUpgradeSubSpell
-    let id = 0
-    let name = ""
-    let toolTip = ""
-    let toolTipExt = ""
-    let hotkey = "E"
-    let icon = "ReplaceableTextures\\CommandButtons\\BTN"
-    let btnPos = new Pair(0, 0)
-    let levelReq = 6
-    let upgradeReq = ""
-
-    construct(int id, string name, string toolTipExt, Pair<int, int> btnPos, string icon, string hotkey)
-        this.id = id
-        this.name = name
-        this.btnPos = btnPos
-        this.hotkey = hotkey
-        this.icon += icon + ".blp"
-        this.toolTipExt = "Raise your class status to " + COLOR_LIGHT_BLUE.toColorString() + name + "|r. " + toolTipExt
-        this.toolTip =  GENERAL_COLOR.toColorString()+"[" + SPECIAL_COLOR.toColorString() + hotkey + GENERAL_COLOR.toColorString() + "] - " + "Transform into " + name
-
-    function buildSpell()
-        new ChannelAbilityPreset(id, 1, true)
-        ..setName(name)
-        ..setIconNormal(icon)
-        ..setIconResearch(icon)
-        ..presetTooltipNormal(lvl -> toolTip)
-        ..presetTooltipNormalExtended(lvl -> toolTipExt + "|nRequires level {0}.".format(this.levelReq.toString()))
+function ChannelAbilityPreset.getLevelReq() returns string
+    return "<{0},reqLevel>".format(this.getNewId().toRawCode())
+
+
+function ChannelAbilityPreset.setUpgradeSubSpell(string name, string toolTipExt)
+    this.setName(name)
+    this.setTooltipNormal(1,  "Transform into ".color(GENERAL_COLOR) + name.color(GOLD_COLOR))
+    let toolTipSubExt = "Raise your class status to {0}. ".format(name.color(GOLD_COLOR)) + toolTipExt + ("|nRequires level {0}".format(this.getLevelReq())).color(GOLD_COLOR)
+    this.setTooltipNormalExtended(1, toolTipSubExt)
+
+function createUpgradeSpell(int newId) returns ChannelAbilityPreset
+    return new ChannelAbilityPreset(newId, 1, true)
         ..presetFollowThroughTime(lvl -> 0)
-        ..setCooldown(1, 0)
+        ..setAnimationNames("")
+        // I don't want people to spawn the spell when they don't have the level required and have their screen flooded by error message
+        ..setCooldown(1, 5)
```

Comment:
This seems unnecessary. The only person they hurt are themselves and there are time when you actually want to spam the spell, e.g. you're in the middle of the fight and are about to get a last hit.